keymanagementscenarios.md

```diff
@@ -0,0 +1,168 @@
+    - Deployer Admin: In some scenarios a deployer will include a group of users (i.e. teams or enterprises). Admin users (i.e. security administrators) will be responsible for configuring deployment policies, including potentially a set of trusted roots or required signatures. 
+    - Deployer Operator: In some scenarios a publisher will include a group of users (i.e. teams or enterprises). Operators (i.e. developers, deployment hosts) will be responsible for pulling container images from registries, verifying their signatures, and then running them.
+- Repository Owner: In some scenarios a container may be stored in a repository that is managed by the repository owner. The repository may be public or private and could also be air-gapped.
+
+## Definitions:
+- Root key: A self signed key used for the lowest designation of trust. Root keys can be created by developers, organizations, public/private CAs, and registry operators. The root key should be retrieved from a trusted source that can establish the authenticity of the creator's identity.
+- Signing key: A signing key is used to generate artifact signatures. A signing key should be signed with a root key or one of its intermediaries. The certificate chain with a signing key can be used to verify which root it belongs to. While a root key can be used as a signing key, this is not recommended as it creates a large blast radius and increases the risk of compromising a root key. 
+- Trust Store: The trust store defines the relationship between signing keys and artifacts that are used at validation time to determine whether to trust an artifact with a crytptographically valid signature. The trust store will relate a source (any source, specific registry, specific repository, or specific target) with a certificate (for root key, intermediate, or signing key) or key repository (for automated key distribtuion). It is not recommended to use a signing key as this will cause signature validation to fail if the signing key is rotated. 
+- Trust Store Certificate: The certificate in the trust store will contain:
+    - Public Key: Will be matched against CRL and signature artifact.
+    - Signed CRL URL: Location to retrieve CRL from.
+    - Signed Identity: Identity of creator of the signed certificate.
+    - (Optional) Issuer: If the certificate is an intermediate this will describe the issuer of the certificate.
+
+## Requirements
+- Signing an artifact SHOULD NOT require the publisher to perform additional actions with a registry/repository or registry operator beyond those required to push an unsigned artifact.
+- Validating a signature SHOULD NOT require the deployer to perform additional actions with a registry/repository or registry operator beyond those required to pull an unsigned artifact.
+- Moving an artifact from one repository to another SHOULD NOT invalidate the signature on the artifact.
+- A rotation of the root key SHOULD NOT require the use of the existing root key.
```

Comment:
I strongly object to making this a requirement, as the language actively discourages users who want to use existing root keys to transparently rotate root keys without using an out-of-band mechanism.

Reply:
Agreed. After watching today's recording, I don't see the value of blocking in-band rotation of root keys. The attacker in that scenario would have the current root key, could sign anything, could create intermediate keys, etc. If they also have access to rotate the root I'm not following how much worse that is since the old root is compromised, and in both situations an out-of-band update is needed to deploy a new trusted root. Conversely the ability to rotate root by a trusted user gives the ability to rotate to a stronger key, transfer control to another authority (department transfers to a central security team), and other useful situations that would be helpful to not force an out-of-band procedure.

Reply:
We should resolve this conversation in the next iteration of requirements.